File: algorithms/strings/is_unique/is_unique.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

bool is_unique(string s)
{
    if (s.length() > 128)
        return false;

    vector<bool> char_set(128, false);

    for (char ch : s)
    {
        if (char_set[(int)ch])
            return false;

        char_set[(int)ch] = true;
    }

    return true;
}

int main()
{
    vector<string> words{"abcde", "hello", "apple", "kite", "padle"};

    for (string word : words)
        cout << word << " : " << (is_unique(word) ? "true" : "false") << endl;

    return 0;
}

File: algorithms/strings/is_unique/is_unique.py
def is_unique(s):
    if len(s) > 128:
        return False

    char_set = [False] * 128

    for ch in s:
        if char_set[ord(ch)]:
            return False

        char_set[ord(ch)] = True

    return True


words = ["abcde", "hello", "apple", "kite", "padle"]

for word in words:
    print("{} : {}".format(word, is_unique(word)))
